### PR TITLE
fix: improve usability of SDK [CFG-1184]

### PR DESCRIPTION
**Pull request by teodora-sandu**
### What this does

This PR includes some improvements from usability testing, split by commit:
- templating more than just Terraform fixture files: Terraform, JSON, YAML
- improving the help docs for the push command so that it's clear what we need to provide
- adding a small comment for the `msg` field in the rule definition
- adding error handling for when the user is calling the `build` and `test` commands with the wrong directory
Screenshot 2021-11-10 at 13 10 41
Screenshot 2021-11-10 at 13 10 49

### Notes for the reviewer

_Instructions on how to run this locally, background context, what to review, questions…_

### More information

- [Jira ticket CFG-1184](https://snyksec.atlassian.net/browse/CFG-1184)

### CODE REVIEW

File: cmd/push.go [hunk 1]
```diff
@@ -17,11 +17,16 @@ var pushCommand = &cobra.Command{
 	Long: `Push a custom rules bundle to an OCI registry.
 
 To start, log into your OCI registry locally and then run:
-$ snyk-iac-rules push <path to generated bundle> --registry <container registry>
+$ snyk-iac-rules push <path to generated bundle> --registry <desired location of OCI artifact in your OCI registry>
 If the tag is not provided at the end of the provided registry, the tool defaults to 'latest'.
 
-The 'push' command takes in the container registry and pushes the bundle at the provided 
-path. The container registry must support OCI artifacts.
+For example, if your generated bundle is called bundle.tar.gz and the OCI registry you're using is DockerHub, the following command must be run:
```
Comment (rontalx) [inline cmd/push.go:23]: extremely minor nit:
maybe we should format the bundle.tar.gz as \`bundle.tar.gz\`?

File: cmd/root.go [hunk 2]
```diff
@@ -27,9 +27,15 @@ $ snyk-iac-rules test --help
...
 
-5. Run the following command to learn how to push the bundle to a Container Registry:
+5. Verify your bundle locally using the Snyk IaC CLI:
+$ snyk iac test --rules --help
+
+6. Run the following command to learn how to push the bundle to an OCI Registry:
 $ snyk-iac-rules push --help
 
+7. Use the bundle stored in the OCI registry in the Snyk IaC CLI:
+$ snyk --help
```
Comment (rontalx) [inline cmd/root.go:37]: is the general snyk cli help command is the right one here?
`snyk --help`

The help commands above looked more specific.

File: cmd/build.go [hunk 3]
```diff
@@ -46,6 +46,16 @@ https://docs.snyk.io/products/snyk-infrastructure-as-code/custom-rules/getting-s
 		}
 		return nil
 	},
+	PostRunE: func(cmd *cobra.Command, args []string) error {
```
Comment (rontalx) [inline cmd/build.go:49]: what does the capital `E` stand for?

**Review by rontalx — APPROVED** [review]
LGTM!
Left a few minor comments

File: cmd/build.go (hunk 3 above)
Comment (teodora-sandu) [reply cmd/build.go:49]: Based on https://github.com/spf13/cobra/blob/master/user_guide.md#returning-and-handling-errors, `E` means that the command can return an error to the user. We are just printing so I changed this to `PostRun` - good question!

File: cmd/root.go (hunk 2 above)
Comment (teodora-sandu) [reply cmd/root.go:37]: If you run `snyk --help` you will get instructions on how to run this with custom rules. Here's the output:
```
   Infrastructure as Code (IAC) scanning
       See snyk iac --help for more details and examples:


           $ snyk iac test /path/to/cloudformation_file.yaml
           $ snyk iac test /path/to/kubernetes_file.yaml
           $ snyk iac test /path/to/terraform_file.tf
           $ snyk iac test /path/to/tf-plan.json
           $ snyk iac test /path/to/arm_file.json



       To use your own custom rules to scan IaC configuration files,  download
       the  snyk-iac-rules  SDK  from  https://github.com/snyk/snyk-iac-rules.
       Follow the instructions there to write, build, and push a custom  rules
       bundle  and  then either use the Snyk UI to configure your custom rules
       settings or configure a remote OCI registry locally by running the fol-
       lowing commands:


           $ snyk config set oci-registry-url=https://registry-1.docker.io/username/repo:tag
           $ snyk config set oci-registry-username=username
           $ snyk config set oci-registry-password=password
```

**Pull request merged**